### PR TITLE
adding optional chaining operator

### PR DESCRIPTION
**Pull request by npeeples**
https://github.com/tc39/proposal-optional-chaining
The only thing that could be bad about this is that 'x?.3:0' is
apparently allowed as a ternary operator. I'm not really sure if that
applies to the syntax highlighter, and is really not a great statement
anyway.

### CODE REVIEW

**Comment by bounceme** [comment]
> (the ?. token must not be immediately followed by a decimal digit)

thanks for letting us know, will add the indentation parsing updates for this

**Comment by bounceme** [comment]
here is the regex: `?\%(\.\d\@!\)\@!`

File: syntax/javascript.vim [hunk 1]
```diff
@@ -162,7 +163,7 @@ syntax region  jsBlock                        matchgroup=jsBraces              s
 syntax region  jsModuleGroup        contained matchgroup=jsModuleBraces        start=/{/ end=/}/   contains=jsModuleKeyword,jsModuleComma,jsModuleAs,jsComment,jsFlowTypeKeyword skipwhite skipempty nextgroup=jsFrom fold
 syntax region  jsSpreadExpression   contained matchgroup=jsSpreadOperator      start=/\.\.\./ end=/[,}\]]\@=/ contains=@jsExpression
 syntax region  jsRestExpression     contained matchgroup=jsRestOperator        start=/\.\.\./ end=/[,)]\@=/
-syntax region  jsTernaryIf                    matchgroup=jsTernaryIfOperator   start=/?/  end=/\%(:\|[\}]\@=\)/  contains=@jsExpression extend skipwhite skipempty nextgroup=@jsExpression
+syntax region  jsTernaryIf                    matchgroup=jsTernaryIfOperator   start=/?[^\.]/  end=/\%(:\|[\}]\@=\)/  contains=@jsExpression extend skipwhite skipempty nextgroup=@jsExpression
```
Comment (bounceme) [inline syntax/javascript.vim]: will fail at eol

**Comment by npeeples** [comment]
thanks for the feedback. not going to lie, I just threw some things in there to make it work, but I'm not really sure where that regex should go for a regular `?`. there doesn't seem to be anything that's just pulling out a regular question mark, and I'm not familiar enough with how the vim syntax files work to understand if there's something else that should be done.

**Comment by npeeples** [comment]
Ok, I think I'm understanding now. That regex is for matching the start of the ternary operator... wow, that's quite the syntax for regex on these vim files.

**Comment by bounceme** [comment]
I think @amadeus might be able to show us how it's done

**Pull request merged**